build/tasks/RemoveSharedFrameworkDependencies.cs

```diff
@@ -71,15 +77,27 @@ private void FilterDependencies(string targetPath, ISet<string> dependencyToRemo
                         updatedGroups.Add(updatedGroup);
                     }
 
-                    if (dirty)
+                    if (referencesFrameworkOnlyAssembly)
                     {
                         packageBuilder.DependencyGroups.Clear();
                         packageBuilder.DependencyGroups.AddRange(updatedGroups);
 
                         var updatedManifest = Manifest.Create(packageBuilder);
+                        var inMemory = new MemoryStream();
+                        updatedManifest.Save(inMemory);
+                        inMemory.Position = 0;
+                        // Hack the raw nuspec to add the <frameworkReference> dependency
+                        var rawNuspec = XDocument.Load(inMemory, LoadOptions.PreserveWhitespace);
+                        var ns = rawNuspec.Root.GetDefaultNamespace();
+                        var metadata = rawNuspec.Root.Descendants(ns + "metadata").Single();
+                        metadata.Add(
+                            new XElement(ns + "frameworkReferences",
+                                new XElement(ns + "group",
+                                    new XAttribute("targetFramework", NuGetFramework.Parse(SharedFrameworkTargetFramework).GetFrameworkString()),
+                                    new XElement(ns + "frameworkReference", new XAttribute("name", "Microsoft.AspNetCore.App")))));
```

Comment:
Surprised the package builder can't be used here. Would be nice to avoid reaching down into the XML.

Is there an issue open somewhere in the NuGet world on this?